### ISSUE TITLE
Druid doesn't work with Java 9

### ISSUE THREAD

**Issue by pcarrier** [apache/incubator-druid#4907]
Running into:

```
    java.lang.NoSuchMethodError: sun.nio.ch.DirectBuffer.cleaner()Lsun/misc/Cleaner;
        at io.druid.java.util.common.ByteBufferUtils.clean(ByteBufferUtils.java:57)
        at io.druid.java.util.common.ByteBufferUtils.free(ByteBufferUtils.java:40)
        at io.druid.java.util.common.ByteBufferUtils.unmap(ByteBufferUtils.java:52)
        at io.druid.java.util.common.MappedByteBufferHandler.close(MappedByteBufferHandler.java:54)
        at io.druid.java.util.common.io.smoosh.FileSmoosher.add(FileSmoosher.java:132)
        at io.druid.java.util.common.io.smoosh.Smoosh.smoosh(Smoosh.java:65)
        at io.druid.segment.IndexMerger.makeIndexFiles(IndexMerger.java:861)
        at io.druid.segment.IndexMerger.merge(IndexMerger.java:434)
        at io.druid.segment.IndexMerger.persist(IndexMerger.java:178)
        at io.druid.segment.IndexMerger.persist(IndexMerger.java:149)
        at io.druid.segment.IndexMerger.persist(IndexMerger.java:127)
        at io.druid.query.aggregation.AggregationTestHelper.createIndex(AggregationTestHelper.java:471)
        at io.druid.query.aggregation.AggregationTestHelper.createIndex(AggregationTestHelper.java:385)
        at io.druid.query.aggregation.AggregationTestHelper.createIndex(AggregationTestHelper.java:353)
        at io.druid.query.aggregation.AggregationTestHelper.createIndexAndRunQueryOnSegment(AggregationTestHelper.java:324)
```

**Comment by leventov** [comment]
@pcarrier thanks. Could you submit a patch that fixes this problem? We don't have any kind of Java 9 environment yet for tests.

**Comment by drcrallen** [comment]
https://github.com/druid-io/druid/issues/5190 I'm going to close down that one since its newer

**Comment by drcrallen** [comment]
http://openjdk.java.net/jeps/260 related to `jdk.unsupported` package, which looks like it only supports `com.sun.nio.file` items

**Comment by xvrl** [comment]
for this particular problem we can probably mimic what @ijuma did to support both Java 7/8 and 9-11, see https://github.com/apache/kafka/commit/e554dc518eaaa0747899e708160275f95c4e525f

**Comment by xvrl** [comment]
This particular error has been fixed with #7466, however there is still more work to be done to support Java 9 and above. I suggest we track all issues related to newer JDK compatibility as part of #5589
